### PR TITLE
docs(s3-request-presigner): add in getSignedUrl example with and without signed headers

### PR DESCRIPTION
**Pull request by whutchinson98**
### Description

Update documentation providing an example on using `getSignedUrl` method with and without signed headers.

### Testing

N/A - docs

### Additional context

While working on setting up presigned urls with signed headers I noticed there was a lack of documentation regarding how to handle `x-amz-*` headers you wish to have signed. I hope this PR allows for others to find the solution quicker.

### Checklist
- [ ] If you wrote E2E tests, are they resilient to concurrent I/O?
- [ ] If adding new public functions, did you add the `@public` tag and enable doc generation on the package?

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/s3-request-presigner/README.md [hunk 1]
```diff
@@ -91,4 +91,46 @@ to `presigned` is not sufficient to make a request. You need to send the
 server-side encryption headers along with the url. These headers remain in the
 `presigned.headers`
 
+getSignedUrl Example:
```
Comment (trivikr) [inline packages/s3-request-presigner/README.md:94]: Add two sections

```md
### Get Presigned URL with headers that cannot be signed

Description, and reason

Example

### Get Presigned URL with headers that should be signed

Description, and reason

Example
```

File: packages/s3-request-presigner/README.md [hunk 2]
```diff
@@ -91,4 +91,46 @@ to `presigned` is not sufficient to make a request. You need to send the
...
+
+```javascript
+import { PutObjectCommand, S3Client } from '@aws-sdk/client-s3';
```
Comment (trivikr) [inline packages/s3-request-presigner/README.md:97]: The example should be prettified using [prettier playground](https://prettier.io/playground/)

File: packages/s3-request-presigner/README.md [hunk 3]
```diff
@@ -91,4 +91,46 @@ to `presigned` is not sufficient to make a request. You need to send the
...
+
+const s3Client = new S3Client({ region: 'us-east-1' });
+const command = new PutObjectCommand({
+    Bucket: bucket,
+    Key: key,
+});
+
+const presigned = getSignedUrl(this.inner, command, {
+      expiresIn: expiration,
+    });
```
Comment (trivikr) [inline packages/s3-request-presigner/README.md:108]: This is not a simple example, and is copied from some other application code.
The `this.inner` doesn't mean anything here.

**Review by trivikr — CHANGES_REQUESTED** [review]

File: packages/s3-request-presigner/README.md (hunk 3 above)
Comment (whutchinson98) [reply packages/s3-request-presigner/README.md]: sorry about that, updated it to use the s3Client

File: packages/s3-request-presigner/README.md (hunk 2 above)
Comment (whutchinson98) [reply packages/s3-request-presigner/README.md]: Formatted

File: packages/s3-request-presigner/README.md (hunk 1 above)
Comment (whutchinson98) [reply packages/s3-request-presigner/README.md]: Thank you, hopefully the changes are better :)

**Review by whutchinson98 — COMMENTED** [review]
@trivikr I believe I've updated everything as requested. Please let me know if there is any other tweaks you would like me to make.

**Review by kuhe — APPROVED** [review]

**Pull request merged**